dagmc_stats/DagmcFile.py

```diff
@@ -106,6 +109,52 @@ def __set_entityset_ranges(self):
                 self._my_moab_core.get_entities_by_type_and_tag(self.root_set, types.MBENTITYSET,
                                                                 self.dagmc_tags['geom_dim'], [dimension])
 
+    def get_meshset_by_id(self, dim, ids=[]):
+        """Get meshset of the geometry with specified dimension and ids
+
+        inputs
+        ------
+        dim : (Integer or String) Dimension of the meshset. 0: 'node(s)',
+                                1: 'curve(s)', 2: 'surface(s)', 3: 'volume(s)'
+        ids : (Integer) Global ID(s) of the meshset
+
+        outputs
+        -------
+        meshset : meshset of the geometry with given dimension and ids. First,
+                  dim will be checked. If dim is invalid, the root set will be
+                  returned. Then, if id is invalid (empty or not in the given
+                  dim range), all entities with the given dim will be returned.
+        """
+        if isinstance(dim, int) and dim in self.entityset_types.keys():
+            dim = self.entityset_types[dim]
+        #print(list(my_dict.keys())[list(my_dict.values()).index(112)])
+        elif type(dim) == str and dim.lower() in ['node', 'curve', 'surface', 'volume', 'nodes', 'curves', 'surfaces', 'volumes']:
+            dim = dim.lower()
+            if dim[-1] != 's':
+                dim = dim + 's'
+        else:
+            # invalid dim
+            warnings.warn('Invalid dim!')
+            return self.root_set
+
+        # if no id is passed in
+        if len(ids) == 0:
+            return self.entityset_ranges[dim]
+
+        dim_ms = self._my_moab_core.create_meshset()
+        self._my_moab_core.add_entity(dim_ms, self.entityset_ranges[dim])
+        meshset = []
+        for id in ids:
```

Comment:
Not in the python API

Reply:
The python API only support one tag at a time, but appear to support multiple values for that tag:

```

values : iterable of tag values
            an iterable data structure of the tag values to be matched for
            entities returned by the call (see general description for details
            on the composition of this data structure)
```

Reply:
I feel like we tried this though and it didn't work.. maybe @yqin43 confirm again? Does removing the for-loop and running this one time actually work?
`self._my_moab_core.get_entities_by_type_and_tag(dim_ms, types.MBENTITYSET, self.dagmc_tags['global_id'], ids)`

Reply:
@gonuke @kkiesling I tried this but there is error with message '???'
```
dagmc_stats/DagmcFile.py:153: in get_meshset_by_id
    meshset = list(self._my_moab_core.get_entities_by_type_and_tag(dim_ms, types.MBENTITYSET, self.dagmc_tags['global_id'], ids))
pymoab/core.pyx:1613: in pymoab.core.Core.get_entities_by_type_and_tag
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   AssertionError

pymoab/core.pyx:1748: AssertionError
```
It seems to be working for ids with one value but not for list of ids.

Reply:
Hmmm?  How about 
`self._my_moab_core.get_entities_by_type_and_tag(dim_ms, types.MBENTITYSET, self.dagmc_tags['global_id'], [ids])`?

Reply:
@gonuke It will return this error:
```
dagmc_stats/DagmcFile.py:153: in get_meshset_by_id
    meshset = self._my_moab_core.get_entities_by_type_and_tag(dim_ms, types.MBENTITYSET, self.dagmc_tags['global_id'], [ids])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: 'pymoab.tag.Tag' object has no attribute '__getitem__'

pymoab/core.pyx:1618: TypeError
```

Reply:
Ok. I guess we'll leave it for now.  I looked through the pymoab tests and I don't see a test for the case I'm describing, although the code makes it look like it should work